src/Aquarius.Client/TimeSeries/Client/AccessTokenConnection.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Reflection;
+using Aquarius.Helpers;
+using ServiceStack.Logging;
+
+namespace Aquarius.TimeSeries.Client
+{
+    public class AccessTokenConnection : IConnection
+    {
+        private static readonly ILog Log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
+
+        private static readonly bool TraceEnabled;
+
+        private readonly object _syncLock = new object();
+        
+        private IAuthenticator Authenticator { get; }
+        private Action<AccessTokenConnection> ConnectionRemovalAction { get; }
+        private string Hostname { get; }
+        internal int ConnectionCount { get; set; }
+        
+        public string AccessToken { get; private set; }
+
+        public string Token() => AccessToken;
+
+        static AccessTokenConnection()
+        {
+            TraceEnabled = AppSettings.Get("TraceConnectionPool", false);
+        }
+
+        public AccessTokenConnection(
+            string hostname,
+            string accessToken,
+            IAuthenticator authenticator,
+            Action<AccessTokenConnection> connectionRemovalAction)
+        {
+            Hostname = hostname;
+            AccessToken = accessToken;
+            Authenticator = authenticator;
+            ConnectionRemovalAction = connectionRemovalAction;
+            ConnectionCount = 1;
+            
+            Trace("Created");
+            
+            CreateNewSession();
+        }
+
+        private void Trace(string message)
+        {
+            if (!TraceEnabled)
+                return;
+            
+            Log.Info($"{GetHashCode()}: {Hostname}/{Token().Substring(0, 4)}/***: ConnectionCount={ConnectionCount} {message}");
+        }
+        
+        private void CreateNewSession()
+        {
+            Authenticator.Login(AccessToken);
+
+            Trace($"NewSession AccessToken={AccessToken}");
+        }
+
+        private void DeleteCurrentSession()
+        {
+            Trace($"Deleting AccessToken={AccessToken}");
+            
+            Authenticator.Logout();
+            AccessToken = null;
+        }
+
+        public void ReAuthenticate()
+        {
+            throw new NotImplementedException(
+                "Re-authentication not supported for access token-based authentication. Login to acquire a new access token.");
```

Comment:
I get that this is an unsupported feature, but how does the client react when this `NotImplementedException` is thrown due to an automatic ReAuthenticate attempt: https://github.com/AquaticInformatics/aquarius-sdk-net/blob/ff1df8bb96064a8e9a8e5a0e4fb1104949acc38d/src/Aquarius.Client/TimeSeries/Client/AquariusClient.cs#L266

Reply:
Not setting the client reauthentication action when using bearer auth